src/modules/users/services/UserService.ts

```diff
@@ -18,7 +18,7 @@ export class UserService {
   private readonly userRepository: UserRepository
   private readonly userLoader: Loader<User>
 
-  constructor({ userRepository, userLoader }: Dependencies) {
+  constructor({ userRepository, userLoader }: UsersDependencies) {
```

Comment:
🟢 I still think we should avoid this pattern of requiring the whole DI container type. It is better replaced by the specific structure that has only the dependencies this service needs, declared nearby. If DI container has required services - it will fit the type anyways.

Reply:
Discussed during the call, this approach would eliminate the compilation time type-safety of the DI